pkg/logs/config/config.go

```diff
@@ -281,3 +285,26 @@ func TaggerWarmupDuration() time.Duration {
 func AggregationTimeout() time.Duration {
 	return defaultLogsConfigKeys().aggregationTimeout()
 }
+
+func getBackupEndpoint(logsConfig *LogsConfigKeys, main Endpoint) *Endpoint {
```

Comment:
This could return an `Endpoint` struct (instead of a pointer), I think this would make the code a bit more straightforward and a bit more consistent? (the caller would only handle `Endpoint` structs, not pointers)

Reply:
Right now `nil` means no backup endpoint - and that bubbles all the way up to the `Endpoints` struct. While it would be more constant with `Main` I think it would be harder to define the absence of a backup endpoint in a way that makes sense.